hexrd/ui/image_stack_dialog.py

```diff
@@ -208,10 +209,11 @@ def total_frames(self):
         self.ui.total_frames.setText(str(total))
 
     def change_detector(self, det):
-        self.detector = det
-        self.setup_gui()
-        if self.state['apply_to_all']:
-            self.state[det]['search'] = self.ui.search_text.text()
+        if det:
+            self.detector = det
+            self.setup_gui()
+            if self.state['apply_to_all']:
+                self.state[det]['search'] = self.ui.search_text.text()
```

Comment:
I personally would just do:
```python
if not det:
    return
```

to perform an early return rather than having to add extra indentation to the subsequent lines of code. But it's a style thing and doesn't need to be changed for this PR.

Reply:
I just went ahead and changed the check to your suggestion to keep things cleaner.